### DISCUSSION TITLE
Welcome to GetNetStat Discussions!

### DISCUSSION THREAD

**Discussion by TobiasPSP** [TobiasPSP/GetNetStat#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by stefanstranger** [comment]
Hi Tobias,

Thanks for creating this module. Any reason why you used classes to develop this module?

Wouldn't this be a great candidate for PowerShell Crescendo? 
 https://devblogs.microsoft.com/powershell/announcing-powershell-crescendo-preview-1/

Stefan

**Comment by TobiasPSP** [comment]
Hi Stefan!

Didn’t know about Crescendo! 😂
However that’s for wrapping native commands so I could use it wrapping native netstat.exe maybe. My module is using core .NET and imho the easiest and most effective approach is adding it as type.

Or did I get the question wrong? My goal was to use something that is avail on all platforms.

Merry Xmas
Tobias

psconf.eu 2021 - June 1-4, Hannover, Germany: https://psconf.eu

**Reply by stefanstranger** [reply]
No you correctly understood my question. I already thought it had something to do with having it to run on all platforms.

Glad you now are aware of Crescendo 😀

**Comment by glatzert** [comment]
Merry Christmas Tobias,

we recently tried some tools (bloodhound) in an IPv6 only environment and it did not work.
I read parts of the P/Invoke code you use here, and would assume it won't be able to connect in IPv6 also.
The solution was to switch all Address-Families to IPv6 and enable DualStack on the socket used.

While I could not test it, yet, I think you might also run into such problems, if there's no IPv4 for the target.

**Comment by TobiasPSP** [comment]
Interesting point Thomas, let’s test! 
The APIs used here should (theoretically) work woth IPv6, too. Thanks for pointing af this. If anyone can test, please report findings here :-). Thx
